pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/client/Auth0ManagementClient.java

```diff
@@ -538,15 +540,25 @@ private <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResu
                 break;
             }
             if (res.getStatusCode() == 429) {
-                log.error(retryMessage, res.getError());
+                log.error(retryMessage, res.getError() + " " + res.getError().getClass().getName());
                 long wait = backoffMillis * numTries + new Random().nextInt(MAX_JITTER_MILLIS);
+                // if we have more information from auth0 about how long to wait, use it.
+                if (res.getError() instanceof RateLimitException) {
+                    RateLimitException rateLimit = (RateLimitException) res.getError();
+                    long unixTimeAtWhichToRetry = rateLimit.getReset();
+                    long suggestedWaitTime = unixTimeAtWhichToRetry - Instant.now().getEpochSecond();
+                    if (suggestedWaitTime > 0) {
+                        wait = suggestedWaitTime * 1000;
+                    }
+                    log.warn("Hit auth0 rate limit.  Pausing for " + wait + "s based on auth0 headers.");
+                } else {
+                    log.warn("Hit auth0 rate limit.  Pausing for " + wait + "ms");
+                }
```

Comment:
snippet from @zyme from the isolated instance repo. I don't know the repo URL.

Reply:
We might want to put an upper bound on the wait time, just in case our math is off or something goes sideways at auth0.  Maybe something like 10s?  I think appengine threads croak after 60s.

Reply:
Agreed. Set 10 seconds max wait time.